saleor/graphql/account/mutations/base.py

```diff
@@ -338,6 +340,10 @@ def post_save_action(cls, info: ResolveInfo, instance, cleaned_input):
             manager = get_plugin_manager_promise(info.context).get()
             cls.call_event(manager.customer_metadata_updated, instance)
 
+        if user_gift_cards := get_user_gift_cards(instance):
+            if cleaned_input.get("first_name") or cleaned_input.get("last_name"):
```

Comment:
It would be better to reverse these conditions to first check inputs and then `get_user_gift_cards`, since the second hits the database, which won't have to be necessary if the first/last name is not changed. We can save some queries this way.

Reply:
good point